File: CS 431 - Project 2 - VirtualMemorySim/MMU.cpp
#include "MMU.h"



void MMU::processVMAddr(VirtualMemoryAddr vma)
{
	if (vma.getWTM()) 
	{
		// write to memory
	}
	else 
	{
		// output to file
	}
}

MMU::MMU()
{
}


MMU::~MMU()
{
}

File: CS 431 - Virtual Memory Sim(Java)/src/project/helpers/MMU.java
package project.helpers;

import project.helpers.VMAddress;
import project.helpers.OutPut;

public class MMU {
    public MMU(){
        outPut = new OutPut();
    }

    void processVMAddr(VMAddress vma){
        if(vma.getWriteToMemory()){

        }
        else{
            outPut.OutPutToFile(vma);
        }
    }
    public OutPut outPut;
}
